lib/remoteproc/remoteproc_virtio.c

```diff
@@ -16,6 +16,47 @@
 #include <metal/utilities.h>
 #include <metal/alloc.h>
 
+static int rproc_virtio_create_virtqueues(struct virtio_device *vdev,
+					  unsigned int flags,
+					  unsigned int nvqs,
+					  const char *names[],
+					  vq_callback callbacks[])
+{
+	struct virtio_vring_info *vring_info;
+	struct vring_alloc_info *vring_alloc;
+	unsigned int num_vrings, i;
+	int ret;
+	(void)flags;
+
+	num_vrings = vdev->vrings_num;
+	if (nvqs > num_vrings)
+		return ERROR_VQUEUE_INVLD_PARAM;
+	/* Initialize virtqueue for each vring */
+	for (i = 0; i < nvqs; i++) {
+		vring_info = &vdev->vrings_info[i];
+
+		vring_alloc = &vring_info->info;
+#ifndef VIRTIO_DEVICE_ONLY
+		if (vdev->role == VIRTIO_DEV_DRIVER) {
+			size_t offset;
+			struct metal_io_region *io = vring_info->io;
+
+			offset = metal_io_virt_to_offset(io,
+							 vring_alloc->vaddr);
+			metal_io_block_set(io, offset, 0,
+					   vring_size(vring_alloc->num_descs,
+						      vring_alloc->align));
+		}
+#endif
+		ret = virtqueue_create(vdev, i, names[i], vring_alloc,
+				       callbacks[i], vdev->func->notify,
+				       vring_info->vq);
+		if (ret)
+			return ret;
+	}
+	return 0;
+}
+
```

Comment:
I can not see something related to remoteproc virtio transport in function above, that why  I need to understand more in details what would be the delta with a virtio MMIO

Reply:
Please reference this pr: https://github.com/OpenAMP/open-amp/pull/494